Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -22,31 +34,845 @@ message EventTypeQuery{
+    
+    google.protobuf.Timestamp ExecutionStop = 5;
+}
+
+message ProcessPipExecutionPerformance_XLGpp{
+    google.protobuf.Duration ProcessExecutionTime = 1;
+    
+    IOTypeCounters_XLGpp ReadCounters = 2;
+
+    IOTypeCounters_XLGpp WriteCounters = 3;
+
+    IOTypeCounters_XLGpp OtherCounters = 4;
+    
+    google.protobuf.Duration UserTime = 5;
+
+    google.protobuf.Duration KernelTime = 6;
+
+    uint64 PeakMemoryUsage = 7;
+
+    int32 PeakMemoryUsageMb = 8;
+
+    uint32 NumberOfProcesses = 9;
+
+    FileMonitoringViolationCounters_XLGpp FileMonitoringViolationCounters = 10;
+
+    Fingerprint_XLGpp Fingerprint = 11;
+
+    uint64 CacheDescriptorId = 12;
+}
+
+message FileMonitoringViolationCounters_XLGpp{
+    int32 NumFileAccessesWhitelistedButNotCacheable = 1;
+
+    int32 NumFileAccessesWhitelistedAndCacheable = 2;
+
+    int32 NumFileAccessViolationsNotWhitelisted = 3;
+}
+
+message Fingerprint_XLGpp{
+    int32 Length = 1;
+
+    bytes Bytes = 2;
+}
+
+message IOTypeCounters_XLGpp{
+    uint64 OperationCount = 1;
+
+    uint64 TransferCOunt = 2;
+}
+
+message ReportedProcess_XLGpp{
+    string Path = 1;
+
+    uint32 ProcessId = 2;
+
+    string ProcessArgs = 3;
+
+    IOTypeCounters_XLGpp ReadCounters = 4;
+
+    IOTypeCounters_XLGpp WriteCounters = 5;
+
+    IOTypeCounters_XLGpp OtherCounters = 6;
+
+    google.protobuf.Timestamp CreationTime = 7;
+
+    google.protobuf.Timestamp ExitTime = 8;
+
+    google.protobuf.Duration KernelTime = 9;
+
+    google.protobuf.Duration UserTime = 10;
+
+    uint32 ExitCode = 11;
+
+    uint32 ParentProcessId = 12;
+}
+
+message ReportedFileAccess_XLGpp{
+    CreationDisposition_XLGpp CreationDisposition = 1;
+
+    DesiredAccess_XLGpp DesiredAccess = 2;
+
+    uint32 Error = 3;
+
+    uint64 Usn = 4;
+
+    FlagsAndAttributes_XLGpp FlagsAndAttributes = 5;
+
+    string Path = 6;
+
+    string ManifestPath = 7;
+
+    ReportedProcess_XLGpp Process = 8;
+
+    ShareMode_XLGpp ShareMode = 9;
+
+    FileAccessStatus_XLGpp Status = 10;
+
+    FileAccessStatusMethod_XLGpp Method = 11;
+
+    RequestedAccess_XLGpp RequestedAccess = 12;
+
+    ReportedFileOperation_XLGpp Operation = 13;
+
+    bool ExplicitlyReported = 14;
+
+    string EnumeratePattern = 15;
+}
+
+message ProcessDetouringStatusData_XLGpp{
+    uint64 ProcessID = 1;
+
+    uint32 ReportStatus = 2;
+
+    string ProcessName = 3;
+
+    string StartApplicationName = 4;
+
+    string StartCommandLine = 5;
+
+    bool NeedsInjection = 6;
+
+    uint64 Job = 7;
+
+    bool DisableDetours = 8;
+
+    uint32 CreationFlags = 9;
+
+    bool Detoured = 10;
+
+    uint32 Error = 11;
+}
+
+message WeakContentFingerPrint_XLGpp{
+    Fingerprint_XLGpp Hash = 1;
+}
+
+message StrongContentFingerPrint_XLGpp{
+    Fingerprint_XLGpp Hash = 1;
+}
+
+message ObservedPathEntry_XLGpp{
+    AbsolutePath_XLGpp Path = 1;
+
+    string EnumeratePatternRegex = 2;
+}
+
+message IUnsafeSandboxConfiguration_XLGpp{

```

Comment:
>IUnsafeSandboxConfiguration_XLGpp [](start = 8, length = 33)

what's the reason for `I` prefix?

Reply:
That is how it was in the BXL codebase, so i simply copied the name over here as well and appended the XLGpp suffix

---
In reply to: [308895911](https://github.com/microsoft/BuildXL/pull/671#discussion_r308895911) [](ancestors = 308895911)